src/pop_verifier.erl

```diff
@@ -1,51 +1,20 @@
-%% @doc code that handles running a full verifier node.
-
 -module(pop_verifier).
+-behavior(gen_server).
 
 -export([
-	 new/3,
-	 on_net_message/5,
-	 on_timer/2,
-	 on_new_block/2,
-	 start_loop/5
+	 init/1, handle_call/3, handle_cast/2, handle_info/2, code_change/3, terminate/2,
+	 make_verifier_array_from_json/1
 	]).
 
 -include_lib("eunit/include/eunit.hrl").
 -include_lib("stdlib/include/assert.hrl").
 
 -include("potato_records.hrl").
 
-
-%% @doc Create new verifier. 
-%% 
-%% Needs to hook up PopManagerConfig.on_new_block to pop_verifier:on_new_block
-
-new(PopConfigData, PopManagerConfig, PopVerConfig) ->
-    #pop_verifier{
-       subscribers = maps:new(),
-       pop_manager = pop_manager:new(PopConfigData, PopManagerConfig),
-       config = PopVerConfig,
-       verifiers_arr = PopConfigData#pop_config_data.verifiers_arr
-      }.
-
-%% @doc Adds/updates a subscriber.
-%% 
-%% Subscribers are sent updates on when new blocks are added.
-%% Each sub expired after a fixed time, to stay subscribed this message needs to be spammed periodically.
-%% (Similar to ping)
-
-on_net_message(SenderAddress, CurrentTime, subscribe, _, PopVerifier) ->
-    Subs = PopVerifier#pop_verifier.subscribers,
-
-    PopVerifier#pop_verifier{subscribers = maps:put(SenderAddress, CurrentTime, Subs)};
-
-on_net_message(SenderAddress, CurrentTime, MsgId, Data, PopVerifier) ->
-    PM0 = PopVerifier#pop_verifier.pop_manager,
-
-    PM1 = pop_manager:on_net_message(SenderAddress, CurrentTime, MsgId, Data, PM0),
-    
-    PopVerifier#pop_verifier{pop_manager = PM1}.
-
+make_event(Code, Data, State) ->
```

Comment:
name of this function is confusing

Reply:
also comments, I guess I"m confused by this event thing in general. Why does pop_verifier need a event handler? Isn't that just suppose to be what handle_info does?

Reply:
not a handler, it has a way of triggering events

someone else would be in charge of processing triggered events